thali/thalireplicationmanager.js

```diff
@@ -192,11 +202,44 @@ function muxServerBridge(tcpEndpointServerPort) {
   var serverPlex = multiplex({}, function(stream, id) {
     var clientSocket = net.createConnection({port: tcpEndpointServerPort});
     stream.pipe(clientSocket).pipe(stream);
+
+    clientSocket.on('end', function () {
+      clientSocket.destroy();
```

Comment:
Why not just write a quick test to be sure? E.g. open a socket, destroy it and then destroy it again. It's best to be sure.